Vagrantfile

```diff
@@ -0,0 +1,131 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+  # The most common configuration options are documented and commented below.
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://vagrantcloud.com/search.
+  config.vm.box = "hashicorp/bionic64"
+
+  #Pyenv setup
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+    # Install pyenv prerequisities
+    sudo apt-get -y update
+    sudo apt install -y make build-essential libssl-dev zlib1g-dev \
+    libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+
+    #Install pyenv
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+
+    cd ~/.pyenv && src/configure && make -C src
+
+    #Configure shell's environment for Python
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+    echo 'eval "$(pyenv init --path)"' >> ~/.profile
+
+    echo -e 'if shopt -q login_shell; then' \
+      '\n  export PYENV_ROOT="$HOME/.pyenv"' \
+      '\n  export PATH="$PYENV_ROOT/bin:$PATH"' \
+      '\n eval "$(pyenv init --path)"' \
+      '\nfi' >> ~/.bashrc
+    echo -e 'if [ -z "$BASH_VERSION" ]; then'\
+      '\n  export PYENV_ROOT="$HOME/.pyenv"'\
+      '\n  export PATH="$PYENV_ROOT/bin:$PATH"'\
+      '\n  eval "$(pyenv init --path)"'\
+      '\nfi' >>~/.profile
+
+    #Adding pyenv to shell
+    echo 'eval "$(pyenv init -)"' >> ~/.bashrc
+
+    #Install Python
+    pyenv install 3.9.2
```

Comment:
On the first run, this step fails because pyenv hasn't yet been picked up on the PATH - to resolve that we should update this shell with the settings we've establish for future ones; we can do that by using `source` to execute the .profile file that we've updated - `source ~/.profile`  - before this line